src/UnderlineNav2/UnderlineNav.tsx

```diff
@@ -0,0 +1,178 @@
+1. Convert context to have array of all items = done
+2. Convert UnderlineNav.Link to ActionList.Item - done
+3. Calculate combined width of items to be 50% of total width available - done
+4. fix the dropdown button - done
+5. figure out horizontal scroll - done
+6. Overflow on resize - done
+7. pass typescript - done
+8. Performance is wack! Resize Observer exceeds limit - done
+9. Test for long strings not wrapping sometimes. - done
+10. Test for mobile device support. - seems good
+11. Add api for Overflow button? - maybe later
+12. Do we need icons in overflow dropdown? Currently not supported. - not doing
+13. Test out and fix selection and highlight - A11y q about hover state.
+14. Write out a proper storybook with tab selection using this nav.
+15. Write up documentation
+16. Demo video.
+*/
+
+type Overflow = 'auto' | 'menu' | 'scroll'
+type ChildWidthArray = Array<{width: number}>
+type ResponsiveProps = {
+  items: Array<React.ReactElement>
+  actions: Array<React.ReactElement>
+}
+const overflowEffect = (
+  newRef: MutableRefObject<HTMLElement>,
+  childArray: Array<React.ReactElement>,
+  childWidthArray: ChildWidthArray,
+  callback: (props: ResponsiveProps) => void
+) => {
+  const domRect = newRef.current.getBoundingClientRect()
+  const width = domRect.width || 0
+
+  if (childWidthArray.length === 0) {
+    callback({items: childArray, actions: []})
+  }
+
+  // do this only for overflow
+  const numberOfItemsPossible = calculatePossibleItems(childWidthArray, width)
+  const items: Array<React.ReactElement> = []
+  const actions: Array<React.ReactElement> = []
+
+  // eslint-disable-next-line github/array-foreach
+  childArray.forEach((child: React.ReactElement, index: number) => {
+    if (index < numberOfItemsPossible) {
+      items.push(child)
+    } else {
+      actions.push(child)
+    }
+  })
+  callback({items, actions})
+}
+
+export type {ResponsiveProps}
+
+function getValidChildren(children: React.ReactNode) {
+  return React.Children.toArray(children).filter(child => React.isValidElement(child)) as React.ReactElement[]
+}
+
+function calculatePossibleItems(childWidthArray: ChildWidthArray, width: number) {
+  let breakpoint = childWidthArray.length - 1
+  let sumsOfChildWidth = 0
+  for (const [index, childWidth] of childWidthArray.entries()) {
+    if (sumsOfChildWidth > 0.5 * width) {
+      breakpoint = index
+      break
+    } else {
+      sumsOfChildWidth = sumsOfChildWidth + childWidth.width
+    }
+  }
+  return breakpoint
+}
+
+export type UnderlineNavProps = {
+  label: string
+  as: React.ElementType
+  overflow: Overflow
+  align?: 'right'
+  sx: SxProp
+  children: React.ReactNode
+}
+
+export const UnderlineNav = forwardRef(
+  ({as = 'nav', overflow = 'auto', align, label, sx: sxProp = {}, children}: UnderlineNavProps, forwardedRef) => {
+    const backupRef = useRef<HTMLElement>(null)
+    const newRef = forwardedRef ?? backupRef
+    const flexDirection = align === 'right' ? 'row-reverse' : 'row'
```

Comment:
Yeah! thanks for the feedback. I'm really not sure why this reverse align feature needs to exist. I only added it because prev component has it. 
Also should that "More" dropdown reverse direction or not? Do you have any design input here?

Reply:
I think the "More" dropdown should be next to the last tab like it is in my screenshot.